### PR TITLE
Enable forecast-only experiments on Hercules

### PR DESCRIPTION
**Pull request by DavidHuber-NOAA**
# Description
This add forecast-only support for Hercules to the global workflow. Partially satisfies #1588.

Refs #1588 

# Type of change
- New feature (adds functionality)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? YES

# How has this been tested?
Forecast-only ATM and ATMA experiments on Hercules.

# Checklist
- [x] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

File: parm/config/gfs/config.aero [hunk 1]
```diff
@@ -11,7 +11,7 @@ case ${machine} in
   "HERA")
     AERO_INPUTS_DIR="/scratch1/NCEPDEV/global/glopara/data/gocart_emissions"
     ;;
-  "ORION")
+  "ORION"|"HERCULES")
```
Comment (WalterKolczynski-NOAA) [inline parm/config/gfs/config.aero]:
```suggestion
  "ORION" | "HERCULES")
```

File: ush/detect_machine.sh [hunk 2]
```diff
@@ -26,6 +26,9 @@ case $(hostname -f) in
 
   Orion-login-[1-4].HPC.MsState.Edu) MACHINE_ID=orion ;; ### orion1-4
 
+  hercules-login-[1-4].hpc.msstate.edu) MACHINE_ID=hercules ;; ### hercules1-4
+  Hercules-login-[1-4].HPC.MsState.Edu) MACHINE_ID=hercules ;; ### hercules1-4
```
Comment (WalterKolczynski-NOAA) [inline ush/detect_machine.sh]: are both of these needed?

File: versions/run.hercules.ver [hunk 3]
```diff
@@ -0,0 +1,10 @@
+export stack_intel_ver=2021.9.0
+export stack_impi_ver=2021.9.0
+
+export ncl_ver=6.6.2
+
+source "${HOMEgfs:-}/versions/run.spack.ver"
+
+# wgrib2 and cdo are different on Hercules from all the other systems
+export wgrib2_ver=3.1.1
+export cdo_ver=2.2.0
```
Comment (WalterKolczynski-NOAA) [inline versions/run.hercules.ver:10]: Is it possible to update the versions for the other machines so they match again?

**Review by WalterKolczynski-NOAA — DISMISSED** [review]
Looks pretty good. There are some things that match what is already there but probably shouldn't be what we are doing that led me to opening a couple new issues to be handled separately.

File: versions/run.hercules.ver (hunk 3 above)
Comment (DavidHuber-NOAA) [reply versions/run.hercules.ver:10]: Unfortunately not.  These versions are not available on the other machines under spack-stack.  It looks like the standard versions (`cdo/2.0.5` and `wgrib2/2.0.8`) [don't compile on Hercules](https://github.com/JCSDA/spack-stack/blob/8780bcdff613e970e2cf881301587b72bd7f3ef8/configs/sites/hercules/packages.yaml#L24-L30).  There are plans to upgrade to these versions for all systems in future releases (https://github.com/JCSDA/spack-stack/pull/884 and https://github.com/JCSDA/spack-stack/issues/859).

File: ush/detect_machine.sh (hunk 2 above)
Comment (DavidHuber-NOAA) [reply ush/detect_machine.sh]: Probably not.  Hercules' nodes are all undercased, but Orion's start with uppercase.  This switched on S4 once after an update, so I put both in here for the just-in-case scenario.  I'm fine with taking it out if that's preferred.

Comment (KateFriedman-NOAA) [reply ush/detect_machine.sh]: FYI Orion once changed case on part of the system as well, which broke our builds, We complained to the Orion helpdesk and got them to change it back (it was an unintentional change IIRC). So we can likely go with what it is now on Hercules.

**Review by KateFriedman-NOAA — APPROVED** [review]
Looks great! No comments. Thanks for this @DavidHuber-NOAA !

File: ush/detect_machine.sh (hunk 2 above)
Comment (DavidHuber-NOAA) [reply ush/detect_machine.sh]: Thanks @KateFriedman-NOAA!  I'll just go with lower case then.

**Review by aerorahul — APPROVED** [review]
Looks good.
Thanks for this first step on porting to Hercules.

**Comment by DavidHuber-NOAA** [comment]
Starting CI testing.

**Comment by junwang-noaa** [comment]
@DavidHuber-NOAA @aerorahul Just let you know we found some issues on HDF5/nccmp on Hercules, see ufs WM issue https://github.com/ufs-community/ufs-weather-model/issues/2015

**Comment by DavidHuber-NOAA** [comment]
Thanks for the heads up @junwang-noaa!

**Pull request merged**